src/adjacent rooms.js

```diff
@@ -0,0 +1,124 @@
+/*
+🏘️
+@file adjacent rooms
+@summary Auto-move between "adjacent" rooms by walking to room edges.
+@license MIT
+@author Violgamba (Jon Heard)
+
+@description
+Auto-move between "adjacent" rooms by walking to room edges.  This plugin is useful if your game
+is built around a grid of rooms, since it allows room-to-room navigation based on room adjacency,
+rather than setting up exit events between each room.  Room adjacencies are defined by a grid-map of
+room ids in the "room-map" text field of this plugin.  See comments for that field below.
+
+NOTE - This plugin IS compatible with the "smooth move" plugin, but must come after it to work.  Use
+the "plugin-order" field for this.
+
+
+HOW TO USE:
+1. Import this plugin into a game that has "adjacent" rooms with unblocked adjacent edges.
+2. Fill this plugin's "room-map" field as explained in the field's comments below.  This defines room adjacencies.
+3. Playtest the game.  Walk to an edge with an adjacent room, then bump it.  Note that you move into the adjacent room.
+
+
+// "room-map" is a text field to hold a grid of room ids layed out with spaces and newlines.  This
+// field lets one define room adjacencies as a grid-map.  Empty grid cells are defined by
+// non-number, non-space characters.
+// I suggest aligning room id's, either with leading 0's or extra spaces.  I also suggest using
+// periods or period clusters for empty grid cells, though any non-numeric character should work.
+// See example below.
+// If your game needs multiple maps to tie all the rooms together then you can add multiple fields
+// named "room-map" to this plugin.  When there are adjacency conflicts between these maps, later
+// maps take precidence.
+//
+// EXAMPLE:
+//
+// 02 01
+// .. 03
+//
+// (01 is the starting room.  To the left of it is 02.  Below it is 03.)
+//
+//!CONFIG room-map (text) ""
+
+
+// If this tag is set, print the array of room adjacencies to the console at playback start.  This
+// is useful to be sure that all adjacencies are properly setup.
+// 
+//!CONFIG log-adjacencies (tag) true
+*/
+
+const roomAdjacencies = {};
+
+function includeAdjacenciesFromMapString(mapString) {
+	if (!mapString.trim()) return;
+	// Split the room-map into lines
+	let roomMap = mapString.split('\n').filter(Boolean);
+	// Split each line into individual cells
+	roomMap = roomMap.map(line => {
+		// Split each line into cells of text (either a number cluster or a non-number cluster)
+		const cells = line.match(/[0-9]+|[^\s0-9]+/g);
+		// Turn each cell of text into either a number or null
+		return cells.map(item => {
+			return parseInt(item, 10) || null;
+		});
+	});
+	// Turn the cell data into adjacency lists
+	for (let y = 0; y < roomMap.length; y++) {
+		for (let x = 0; x < roomMap[y].length; x++) {
+			const roomId = roomMap[y][x];
+			// NOTE: map id's start at 1 so falsy checks are fine.
```

Comment:
although bipsi starts map IDs at 1 in the editor, 0 is still a supported map ID, and older games/templates may still use it as an ID. i happen to be on an older PC atm and this caused some issues while i was testing, but seems simply to fix so i'll push an update shortly

Reply:
Thanks for catching this.  i'll keep 0 ids as valid for different object types going forward.